### PR TITLE
IGNITE-12358 Migrate ZeroMQ module to ignite-extensions

### PR DESCRIPTION
**Pull request by samaitra**

### CODE REVIEW

File: modules/zeromq-ext/pom.xml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+    <artifactId>ignite-zeromq-ext</artifactId>
+    <version>1.0.0-SNAPSHOT</version>
+    <url>http://ignite.apache.org</url>
+
+    <properties>
+        <jeromq.version>0.3.6</jeromq.version>
+        <ignite-core.version>2.9.0-SNAPSHOT</ignite-core.version>
+        <ignite-log4j.version>2.9.0-SNAPSHOT</ignite-log4j.version>
+        <ignite-spring.version>2.9.0-SNAPSHOT</ignite-spring.version>
```
Comment (Mikhail1988) [inline modules/zeromq-ext/pom.xml]: it should be a single ignite.version, we do not need to have a separate version for each jar.
Furthermore, why do we have it here? I think it should be defined in parent pom, so all extensions will be built and release for the same ignite version. However, if we want to treat all extensions as absolutely independent unites and each one can support different ignite versions, then why we put all extensions under a single umbrella project?

File: modules/zeromq-ext/README.txt [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    ...
+    <dependencies>
+        ...
+        <dependency>
+            <groupId>org.apache.ignite</groupId>
+            <artifactId>ignite-zeromq</artifactId>
+            <version>${ignite.version}</version>
+        </dependency>
+        ...
+    </dependencies>
```
Comment (Mikhail1988) [inline modules/zeromq-ext/README.txt]: it's not true anymore and needs to be fixed.
but I like idea to match extension version and ignite version, so let's replace 1.0.0-SNAPSHOT with ignite version 2.9.0-SNAPSHOT.

File: modules/zeromq-ext/modules/core/src/test/config/tests.properties [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+ssh.username=uname
+ssh.password=passwd
+
+# SSL tests keystore.
+ssl.keystore.path=@{IGNITE_HOME}/modules/clients/src/test/keystore/server.jks
+ssl.keystore.password=123456
+
+# node01 signed with trust-one, node02 and node03 by trust-two, node02old is expired
+# trust-both contains both CAs
+ssl.keystore.node01.path=@{IGNITE_HOME}/modules/clients/src/test/keystore/ca/node01.jks
```
Comment (alamar) [inline modules/zeromq-ext/modules/core/src/test/config/tests.properties]: Do we need all of these properties here? These files are not reachable from ignite-extensions

File: pom.xml [hunk 4]
```diff
@@ -44,6 +44,7 @@
     <modules>
         <module>modules/flink-ext</module>
         <module>modules/pub-sub</module>
+        <module>modules/zeromq-ext</module>
```
Comment (alamar) [inline pom.xml]: Do we really need to add -ext to all of those modules' names?

File: modules/zeromq-ext/pom.xml (hunk 1 above)
Comment (samaitra) [reply modules/zeromq-ext/pom.xml]: @Mikhail1988 I have updated the PR

File: modules/zeromq-ext/README.txt (hunk 2 above)
Comment (samaitra) [reply modules/zeromq-ext/README.txt]: I have updated the docs

File: modules/zeromq-ext/modules/core/src/test/config/tests.properties (hunk 3 above)
Comment (samaitra) [reply modules/zeromq-ext/modules/core/src/test/config/tests.properties]: yes, I had initially put it under ignite-extensions path but the tests were failing. So, I had put it in a locally resolvable path.

File: pom.xml (hunk 4 above)
Comment (samaitra) [reply pom.xml]: I had put `ext` to avoid conflict with existing released version. As we are planning to release from 1.0.0 version.

**Pull request closed without merging**